scripts/install_deps.py

```diff
@@ -0,0 +1,112 @@
+import os
+import platform
+import shutil
+import subprocess
+import sys
+
+from contextlib import contextmanager
+from utils import _virtualenv_enabled
+
+INSTALL_ARGS = "--no-binary :all: --no-build-isolation --no-cache-dir --no-index "
+
+
+class BadRCError(Exception):
+    pass
+
+
+class InstallationError(Exception):
+    pass
+
+
+@contextmanager
+def cd(dirname):
+    original = os.getcwd()
+    os.chdir(dirname)
+    try:
+        yield
+    finally:
+        os.chdir(original)
+
+
+def run(cmd):
+    sys.stdout.write("Running cmd: %s\n" % cmd)
+    p = subprocess.Popen(
+        cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+    stdout, stderr = p.communicate()
+    if p.returncode != 0:
+        output = (stdout + stderr).decode("utf-8")
+        raise BadRCError("Bad rc (%s) for cmd '%s': %s" % (p.returncode, cmd, output))
+    return stdout
+
+
+def _get_package_tarball(package_dir, package_prefix):
+    package_filenames = sorted(
+        [p for p in os.listdir(package_dir) if p.startswith(package_prefix)]
+    )
+    if len(package_filenames) == 0:
+        raise InstallationError(
+            "Unable to find local package starting with %s prefix."
+            % package_prefix
+        )
+    return package_filenames[-1]
+
+
+def pip_install_packages(package_dir):
+    package_dir = os.path.abspath(package_dir)
+
+    pip_script = os.path.join(os.environ["VIRTUAL_ENV"], "bin", "pip")
+    run("%s install --upgrade pip" % pip_script)
```

Comment:
Given that we are trying to make a reproducible build, and pip has removed arguments before. What are your thoughts on locking this down as well. They also don't follow semver anymore so its difficult to trust a range.

Reply:
Yep agreed, let me find a version of 20 that's stable with the build infra and we'll just pin to that.